newspaper/article.py

```diff
@@ -160,14 +161,26 @@ def download(self, input_html=None, title=None, recursion_counter=0):
         infinite
         """
         if input_html is None:
-            try:
-                html = network.get_html_2XX_only(self.url, self.config)
-            except requests.exceptions.RequestException as e:
-                self.download_state = ArticleDownloadState.FAILED_RESPONSE
-                self.download_exception_msg = str(e)
-                log.debug('Download failed on URL %s because of %s' %
-                          (self.url, self.download_exception_msg))
-                return
+            parsed_url = urlparse(self.url)
+            if parsed_url.scheme == "file":
```

Comment:
Now that the `def download(..)` fn is quite large let's split up the logic of handling `if parsed_url.scheme == "file":` versus scheme http in different helper functions?

Maybe something two new functions in the `Article`  class such as
```
def _parse_scheme_file(parsed_url):
  try:
    with open(parsed_url.path, "r") as fin:
      return fin.read()
  except OSError as e:
    self.download_state = ArticleDownloadState.FAILED_RESPONSE
    self.download_exception_msg = e.strerror
    log.debug('Download failed on URL %s because of %s' %
                      (self.url, self.download_exception_msg))
    return

def _parse_scheme_http():
  try:
    html = network.get_html_2XX_only(self.url, self.config)
  except requests.exceptions.RequestException as e:
    self.download_state = ArticleDownloadState.FAILED_RESPONSE
    ... <and so on>

def download(...):
  ... <bunch of logic>

  if parsed_url.scheme == "file":
    _parse_scheme_file(parsed_url)
  else:
    _parse_scheme_http()

```